[test_demo_vit_ttnn_inference_perf_e2e_2cq_trace.py] | (Single-card) Demo tests (no perf) | single-card-demo-tests / vit-N300-func

## CI Failure Report

<!-- AUTO-TRIAGE-SUMMARY-START -->
**Summary:** [CI] ViT N300 demo test fails with OOM allocating TRACE buffer in end_trace_capture
<!-- AUTO-TRIAGE-SUMMARY-END -->

**Workflow:** (Single-card) Demo tests (no perf)
**Job:** single-card-demo-tests / vit-N300-func

### Regression timeline

- **Last passing:** commit `c50a3fbc41f1` on 2026-06-22T19:12:05Z — https://github.com/tenstorrent/tt-metal/actions/runs/27977468482
- **First failing:** commit `3ecd00d73ca0` on 2026-06-23T13:53:49Z — https://github.com/tenstorrent/tt-metal/actions/runs/28031411077/job/83016560835
- **Most recent failing:** commit `4b39efbe9291` on 2026-06-23T18:49:04Z — https://github.com/tenstorrent/tt-metal/actions/runs/28049149315/job/83081210869

### Failing job URLs (last 2 runs)

- https://github.com/tenstorrent/tt-metal/actions/runs/28049149315/job/83081210869
- https://github.com/tenstorrent/tt-metal/actions/runs/28031411077/job/83016560835

### Failing test path(s)

- `models/demos/vision/classification/vit/wormhole/tests/test_demo_vit_ttnn_inference_perf_e2e_2cq_trace.py::test_vit[8-device_params0]`

### Error excerpt (terminal failure from logs)

**Run 1** (`4b39efbe9291`):
```
E       RuntimeError: TT_FATAL @ /project/tt_metal/impl/allocator/bank_manager.cpp:462: false
E       info:
E       Out of Memory: Not enough space to allocate 1703936 B TRACE buffer across 12 banks, where each bank needs to store 147456 B, but bank size is 146112 B (allocated: 0 B, free: 146112 B, largest free block: 146112 B)

FAILED models/demos/vision/classification/vit/wormhole/tests/test_demo_vit_ttnn_inference_perf_e2e_2cq_trace.py::test_vit[8-device_params0] - RuntimeError: TT_FATAL @ /project/tt_metal/impl/allocator/bank_manager.cpp:462: false
```

**Run 2** (`3ecd00d73ca0`):
```
E       RuntimeError: TT_FATAL @ /project/tt_metal/impl/allocator/bank_manager.cpp:462: false
E       info:
E       Out of Memory: Not enough space to allocate 1703936 B TRACE buffer across 12 banks, where each bank needs to store 147456 B, but bank size is 146112 B (allocated: 0 B, free: 146112 B, largest free block: 146112 B)

FAILED models/demos/vision/classification/vit/wormhole/tests/test_demo_vit_ttnn_inference_perf_e2e_2cq_trace.py::test_vit[8-device_params0] - RuntimeError: TT_FATAL @ /project/tt_metal/impl/allocator/bank_manager.cpp:462: false
```

The failure occurs at `ttnn.end_trace_capture(device, trace_id, cq_id=0)` when the allocator tries to allocate a TRACE buffer but L1 bank size (146112 B) is smaller than what each bank needs to hold (147456 B) for the 1703936 B total trace.

### Reproduction steps

```bash
pytest models/demos/vision/classification/vit/wormhole/tests/test_demo_vit_ttnn_inference_perf_e2e_2cq_trace.py::test_vit -k "8-device_params0" -v
```

Requires an N300 (Wormhole) device. Set environment variables as used in CI:
```
TT_METAL_OPERATION_TIMEOUT_SECONDS=5
TT_TRIAGE_ENABLE_AGGREGATED_CALLSTACKS=1
```

### Notes

- The test uses 2CQ trace mode (`test_demo_vit_ttnn_inference_perf_e2e_2cq_trace.py`) and fails during `end_trace_capture` when the trace buffer allocation requests 147456 B per bank but only 146112 B is available per bank — a shortfall of 1344 B per bank.
- This is an OOM condition on L1 memory, suggesting the trace has grown (or available memory has shrunk) since the last passing commit.
- The regression window is between `c50a3fbc41f1` (passing, 2026-06-22) and `3ecd00d73ca0` (failing, 2026-06-23).
- **Determinism check:**
  - Run 1: `FAILED test_vit[8-device_params0] - RuntimeError: TT_FATAL @ /project/tt_metal/impl/allocator/bank_manager.cpp:462: false` — OOM allocating 1703936 B TRACE buffer, bank size 146112 B < required 147456 B
  - Run 2: `FAILED test_vit[8-device_params0] - RuntimeError: TT_FATAL @ /project/tt_metal/impl/allocator/bank_manager.cpp:462: false` — OOM allocating 1703936 B TRACE buffer, bank size 146112 B < required 147456 B
  - Both runs: identical error class, identical file/line, identical allocation sizes.

<!-- AUTO-TRIAGE-METADATA-START -->
`Auto-triage-workflow: (Single-card) Demo tests (no perf)`
`Auto-triage-job-name: single-card-demo-tests / vit-N300-func`
<!-- AUTO-TRIAGE-METADATA-END -->

<!-- PIPELINE-REPORT-START -->
## Pipeline report
Analysis report: https://github.com/tenstorrent/tt-metal/actions/runs/28066122907
Slack notification: https://tenstorrent.slack.com/archives/C08SJ7MGESY/p1782262886041629?thread_ts=1782260230.462759&cid=C08SJ7MGESY
<!-- PIPELINE-REPORT-END -->

<!-- AUTO-TRIAGE-INFO-START -->
## About this issue (automated CI auto-triage)
This issue was opened and assigned by an automated AI system. It posts through a bot token, so notifications may look like they come from a person, but they are automated.

The owner is selected from recent commit history and job-ownership data. It reflects who is *likely able to fix or route* this, not necessarily who caused it, so it can be imperfect.

**If this was assigned to the wrong person, you can:**
- Comment `REASSIGN OWNER: <why this is outside your area / which team or test-area it belongs to>` and the system will automatically re-route it to a better-suited owner.
- Unassign yourself (this stops notifications to you).
- Close the issue if it is not a real or relevant failure.

The assigned owner is reminded about every 3 days until the issue is resolved. Full guide: https://tenstorrent.atlassian.net/wiki/spaces/MI6/pages/2545221692/Auto-Triage+Info
<!-- AUTO-TRIAGE-INFO-END -->

--- a/models/demos/vision/classification/vit/t3000/demo/demo_vit_performant_imagenet_inference.py
+++ b/models/demos/vision/classification/vit/t3000/demo/demo_vit_performant_imagenet_inference.py
@@ -22,7 +22,7 @@ NUM_VALIDATION_IMAGES_IMAGENET = 49920
 @run_for_wormhole_b0()
 @pytest.mark.model_perf_t3000
 @pytest.mark.parametrize(
-    "device_params", [{"l1_small_size": 32768, "num_command_queues": 2, "trace_region_size": 1700000}], indirect=True
+    "device_params", [{"l1_small_size": 32768, "num_command_queues": 2, "trace_region_size": 2_000_000}], indirect=True
 )
 @pytest.mark.parametrize(
     "mesh_device",
@@ -137,7 +137,7 @@ def test_run_vit_trace_2cqs_inference(
 @run_for_wormhole_b0()
 @pytest.mark.model_perf_t3000
 @pytest.mark.parametrize(
-    "device_params", [{"l1_small_size": 32768, "num_command_queues": 2, "trace_region_size": 1700000}], indirect=True
+    "device_params", [{"l1_small_size": 32768, "num_command_queues": 2, "trace_region_size": 2_000_000}], indirect=True
 )
 @pytest.mark.parametrize(
     "mesh_device",

--- a/models/demos/vision/classification/vit/wormhole/demo/demo_vit_performant_imagenet_inference.py
+++ b/models/demos/vision/classification/vit/wormhole/demo/demo_vit_performant_imagenet_inference.py
@@ -19,7 +19,7 @@ NUM_VALIDATION_IMAGES_IMAGENET = 49920
 
 @run_for_wormhole_b0()
 @pytest.mark.parametrize(
-    "device_params", [{"l1_small_size": 32768, "num_command_queues": 2, "trace_region_size": 1700000}], indirect=True
+    "device_params", [{"l1_small_size": 32768, "num_command_queues": 2, "trace_region_size": 2_000_000}], indirect=True
 )
 @pytest.mark.parametrize(
     "batch_size_per_device, iterations",
@@ -124,7 +124,7 @@ def test_run_vit_trace_2cqs_inference(
 
 @run_for_wormhole_b0()
 @pytest.mark.parametrize(
-    "device_params", [{"l1_small_size": 24576, "trace_region_size": 1700000, "num_command_queues": 2}], indirect=True
+    "device_params", [{"l1_small_size": 24576, "trace_region_size": 2_000_000, "num_command_queues": 2}], indirect=True
 )
 @pytest.mark.parametrize(
     "batch_size_per_device, iterations",

--- a/models/demos/vision/classification/vit/wormhole/tests/test_demo_vit_ttnn_inference_perf_e2e_2cq_trace.py
+++ b/models/demos/vision/classification/vit/wormhole/tests/test_demo_vit_ttnn_inference_perf_e2e_2cq_trace.py
@@ -119,7 +119,13 @@ def run_trace_2cq_model(device, test_infra, num_warmup_iterations, num_measureme
 @pytest.mark.models_performance_bare_metal
 @pytest.mark.models_performance_virtual_machine
 @pytest.mark.parametrize(
-    "device_params", [{"l1_small_size": 32768, "num_command_queues": 2, "trace_region_size": 1753088}], indirect=True
+    # trace_region_size is the TOTAL trace budget across DRAM banks (#47122). The captured
+    # ViT 2CQ trace is ~1.70 MiB; with 12 banks and max page size 8192, each bank needs
+    # 147456 B. 1753088 only reserved 146112 B/bank under pre-#47766 dram_alignment
+    # rounding and OOM'd on N300 (#47887). 2_000_000 leaves headroom after page rounding.
+    "device_params",
+    [{"l1_small_size": 32768, "num_command_queues": 2, "trace_region_size": 2_000_000}],
+    indirect=True,
 )
 @pytest.mark.parametrize("batch_size", [8])
 def test_vit(device, batch_size, is_single_card_n300):

